docs/fides/docs/development/release_checklist.md

```diff
@@ -1,56 +0,0 @@
-# Release Checklist
```

Comment:
seems like we would still want this file?

Reply:
Yes I just copied this file into the PR when I released 2.11.0!

Reply:
Ah, that's my bad. I assumed we were only using the big one in Confluence these days! I'll re-add this and cross-link to the confluence one for clarity as well